src/Microsoft.AspNetCore.Mvc.Razor.Precompilation/Internal/PrecompileRunCommand.cs

```diff
@@ -0,0 +1,193 @@
+            }
+
+            return true;
+        }
+
+        private void UpdateAssembly(string outputPath, PrecompilationResult precompilationResult)
+        {
+            var host = new PeReader.DefaultHost();
+
+            var assemblyPath = Path.Combine(outputPath, Path.GetFileName(ProjectPath) + ".dll");
+            var assembly = host.LoadUnitFrom(assemblyPath) as IAssembly;
+            assembly = new MetadataDeepCopier(host).Copy(assembly);
+
+            var rewriters = new MetadataRewriter[]
+            {
+                new AddResourcesRewriter(host, precompilationResult.CompileOutputs),
+                new RemoveStrongNameRewriter(host)
+            };
+
+            var updatedAssembly = assembly;
+            foreach (var rewriter in rewriters)
+            {
+                updatedAssembly = rewriter.Rewrite(updatedAssembly);
+            }
+
+            using (var stream = File.OpenWrite(Path.ChangeExtension(assemblyPath, ModifiedAssemblyExtension)))
+            {
+                PeWriter.WritePeToStream(updatedAssembly, host, stream);
+            }
+        }
+
+        private PrecompilationResult CompileViews(IFileProvider fileProvider)
+        {
+            var precompilationResult = new PrecompilationResult();
+            foreach (var fileInfo in FileProviderUtilities.GetRazorFiles(fileProvider))
+            {
+                CompileView(precompilationResult, fileInfo);
+            }
+
+            return precompilationResult;
+        }
+
+        private void CompileView(PrecompilationResult result, RelativeFileInfo fileInfo)
+        {
+            GeneratorResults generatorResults;
+            using (var fileStream = fileInfo.FileInfo.CreateReadStream())
+            {
+                generatorResults = Host.GenerateCode(fileInfo.RelativePath, fileStream);
+            }
+
+            if (!generatorResults.Success)
+            {
+                result.RazorErrors.AddRange(generatorResults.ParserErrors);
+                return;
+            }
+
+            var compileOutputs = new CompileOutputs(fileInfo.RelativePath, GeneratePdbOption.HasValue());
+            var emitResult = CompilationService.EmitAssembly(
+                PrecompiledAssemblyPrefix + Guid.NewGuid(),
+                generatorResults.GeneratedCode,
+                compileOutputs.AssemblyStream,
+                compileOutputs.PdbStream);
+
+            if (!emitResult.Success)
+            {
+                result.RoslynErrors.AddRange(emitResult.Diagnostics);
+                compileOutputs.Dispose();
+            }
+            else
+            {
+                result.CompileOutputs.Add(compileOutputs);
+            }
+        }
+
+        private IServiceProvider ConfigureDefaultServices()
+        {
+            var services = new ServiceCollection();
+
+            var applicationName = Path.GetFileName(ProjectPath.TrimEnd('/'));
+            var contentRoot = ContentRootOption.HasValue()
```

Comment:
Ahh, I thought `.Value()` threw when you didn't have a value. I can change this.